src/main/java/cn/nukkit/block/BlockSponge.java

```diff
@@ -61,17 +63,29 @@ public BlockColor getColor() {
     public boolean place(Item item, Block block, Block target, BlockFace face, double fx, double fy, double fz, Player player) {
         Level level = block.getLevel();
         boolean blockSet = level.setBlock(block, this);
-        if (blockSet && this.getDamage() == 0 && performWaterAbsorb(block)) {
-            level.setBlock(block, Block.get(BlockID.SPONGE, 1));
-
-            for (int i = 0; i < 4; i++) {
-                LevelEventPacket packet = new LevelEventPacket();
-                packet.evid = 2001;
-                packet.x = (float) block.getX();
-                packet.y = (float) block.getY();
-                packet.z = (float) block.getZ();
-                packet.data = GlobalBlockPalette.getOrCreateRuntimeId(BlockID.WATER, 0);
-                level.addChunkPacket(getChunkX(), getChunkZ(), packet);
+
+        if (blockSet) {
+            if (this.getDamage() == 1 && level.getDimension() == 1) {
```

Comment:
Maybe use constant Level.DIMENSION_NETHER instead of 1. In case of future changes.

Reply:
Thanks for the idea